docs/ai/quickstarts/text-to-image.md

```diff
@@ -0,0 +1,140 @@
+   The preceding code:
+
+   - Loads configuration from user secrets.
+   - Creates an `ImageClient` from the OpenAI SDK.
+   - Converts the `ImageClient` to an `IImageGenerator` using the <xref:Microsoft.Extensions.AI.OpenAIClientExtensions.AsIImageGenerator(OpenAI.Images.ImageClient)> extension method.
+
+1. Add the following code to implement basic text-to-image generation:
+
+   :::code language="csharp" source="snippets/text-to-image/azure-openai/Program.cs" id="GenerateImage":::
+
+   The preceding code:
+
+   - Generates an image using the <xref:Microsoft.Extensions.AI.ImageGeneratorExtensions.GenerateImagesAsync(Microsoft.Extensions.AI.IImageGenerator,System.String,Microsoft.Extensions.AI.ImageGenerationOptions,System.Threading.CancellationToken)> method with a text prompt.
+   - Saves the generated image to a file in the local user directory.
+
+1. Run the application, either through the IDE or using `dotnet run`.
+
+   The application generates an image and outputs the file path to the image. Open the file to view the generated image. The following image shows one example of a generated image.
+
+   :::image type="content" source="media/text-to-image/jungle-tennis.png" alt-text="AI-generated image of a tennis court in a jungle.":::
+
+## Configure image generation options
+
+You can customize image generation by providing options such as size, response format, and number of images to generate. The <xref:Microsoft.Extensions.AI.ImageGenerationOptions> class allows you to specify:
+
+- <xref:Microsoft.Extensions.AI.ImageGenerationOptions.AdditionalProperties>: Provider-specific options.
+- <xref:Microsoft.Extensions.AI.ImageGenerationOptions.Count>: The number of images to generate.
+- <xref:Microsoft.Extensions.AI.ImageGenerationOptions.ImageSize>: The dimensions of the generated image as a <xref:System.Drawing.Size?displayProperty=fullName>. Supported sizes are 1024 x 1024, 1024 x 1536, and 1536 x 1024.
+- <xref:Microsoft.Extensions.AI.ImageGenerationOptions.MediaType>: The media type (MIME type) of the generated image.
+- <xref:Microsoft.Extensions.AI.ImageGenerationOptions.ModelId>: The model ID.
+- <xref:Microsoft.Extensions.AI.ImageGenerationOptions.RawRepresentationFactory>: The callback that creates the raw representation of the image generation options from an underlying implementation.
+- <xref:Microsoft.Extensions.AI.ImageGenerationOptions.ResponseFormat>: Options are <xref:Microsoft.Extensions.AI.ImageGenerationResponseFormat.Uri>, <xref:Microsoft.Extensions.AI.ImageGenerationResponseFormat.Data>, and <xref:Microsoft.Extensions.AI.ImageGenerationResponseFormat.Hosted>.
+
+Update your code to include configuration options:
+
+:::code language="csharp" source="snippets/text-to-image/azure-openai/Program.cs" id="WithOptions":::
+
+## Handle errors and edge cases
+
+It's important to handle potential errors such as content filtering, rate limiting, or invalid sizes when you generate images. Add error handling to your application:
```

Comment:
The first two might be common to other samples which deal with models / MEAI in general.  The sort of errors that come from IImageClient should be similar to IChatClient.

For things like "invalid sizes" -- these are model/provider specific.  So while it may be helpful in seeing these errors when integrating new models or providers I don't expect those to be "surprise" runtime exceptions that might occur later on.  Of course if the size is exposed as a user settable thing then this might be an exception that needs to be handled.

Reply:
Okay, I'll remove this section.